pandas/core/indexes/multi.py

```diff
@@ -3175,6 +3175,14 @@ def _wrap_joined_index(self, joined, other):
     @Appender(Index.isin.__doc__)
     def isin(self, values, level=None):
         if level is None:
+            # validate value length
```

Comment:
hmm, we already do this sort of validation in ``_verify_integrity`` in the MI construction, so I think the MI is actually getting constructed *correctly*, but you have nan's in the levels.

I am not opposed to the check, but it seems that maybe the semantics of handling this are not exactly right, meaning what you passed in your tests is actually valid.